#ifndef __ccipdProstaFuseOptions_h
#define __ccipdProstaFuseOptions_h





//////////////////////////////////////////////////////////////////////////////////////////

// ccipd includes
#include "ccipdMacros.h"

#ifdef _MSC_VER
#pragma warning( push, 0 )
#endif // _MSC_VER

// qt includes
#include <QDialog>

#ifdef _MSC_VER
#pragma warning( pop )
#endif // _MSC_VER

//////////////////////////////////////////////////////////////////////////////////////////



//////////////////////////////////////////////////////////////////////////////////////////
// Forward Qt class declarations
class Ui_ProstaFuseOptions;
//////////////////////////////////////////////////////////////////////////////////////////
 

namespace ccipd
{


/**
 * \brief Options GUI window for ProstaFuse.
 * 
 * \author  Rachel Sparks
 * \ingroup ProstaFuse
 */
class ProstaFuseOptions : public QDialog
{



//////////////////////////////////////////////////////////////////////////////////////////
  Q_OBJECT
//////////////////////////////////////////////////////////////////////////////////////////



public:
//////////////////////////////////////////////////////////////////////////////////////////
  
  ProstaFuseOptions(
    const bool useAffineTransform  = true,
    const bool useElasticTransform = false,
    const bool cropMovedImage      = false	); /// < Constructor

  virtual ~ProstaFuseOptions(); ///< Destructor

  /// Should you apply the affine transform to the study
  ccipdGetConstDeclareMacro( UseAffineTransform, bool );

  /// Should you apply the elastic transform to the study
  ccipdGetConstDeclareMacro( UseElasticTransform, bool );


  /// Should you apply the elastic transform to the study
  ccipdGetConstDeclareMacro( CropMovedImage, bool );
//////////////////////////////////////////////////////////////////////////////////////////



public slots:
//////////////////////////////////////////////////////////////////////////////////////////
//////////////////////////////////////////////////////////////////////////////////////////



protected:
protected slots:
//////////////////////////////////////////////////////////////////////////////////////////
//////////////////////////////////////////////////////////////////////////////////////////
 


private:
//////////////////////////////////////////////////////////////////////////////////////////
  /// Designer form
  Ui_ProstaFuseOptions * const ui;
//////////////////////////////////////////////////////////////////////////////////////////



//////////////////////////////////////////////////////////////////////////////////////////
  /// PIMPL macro
  //ccipdPIMPLMacro
//////////////////////////////////////////////////////////////////////////////////////////



}; // class
} // namespace
#endif // __ccipdProstaFuseOptions_h
